File: app/src/main/java/in/ceeq/define/firebase/MyFirebaseInstanceIdService.java
package in.ceeq.define.firebase;

import com.google.firebase.iid.FirebaseInstanceId;
import com.google.firebase.iid.FirebaseInstanceIdService;

import in.ceeq.define.utils.PreferenceUtils;

public class MyFirebaseInstanceIdService extends FirebaseInstanceIdService {

    @Override
    public void onTokenRefresh() {
        String refreshedToken = FirebaseInstanceId.getInstance().getToken();
        PreferenceUtils.newInstance(this).setUniqueId(refreshedToken);
    }
}

File: app/src/main/java/in/ceeq/define/firebase/MyFirebaseInstanceIdService.kt
package `in`.ceeq.define.firebase

import `in`.ceeq.define.utils.PreferenceUtils
import android.preference.PreferenceManager
import com.google.firebase.iid.FirebaseInstanceId
import com.google.firebase.iid.FirebaseInstanceIdService

class MyFirebaseInstanceIdService : FirebaseInstanceIdService() {

    override fun onTokenRefresh() {
        val refreshedToken = FirebaseInstanceId.getInstance().token!!
        PreferenceUtils(PreferenceManager.getDefaultSharedPreferences(application)).uniqueId = refreshedToken
    }
}
